ONSdigital/dis-wagtail-alpha
# Add front-end tooling so we can use our own JS and CSS in the repo

### What is the context of this PR?
- This PR adds the ability to write custom JS and CSS separately from the design system
- Uses static_src as the source directory for static files, and static_compiled as the target directory for compiled and minimised CSS, JS and images
- Adds `static_src` to the `STATICFILES_DIRS` array (the current `jinja2/assets` folder also still works)
- Adds linting using torchbox set-up
- Adds all the usual kit set-up with webpack and package.json
- Updates the CI to ensure front-end files are compiled and linted
- Updates pre-commit to lint front-end files
- Adds a temporary rule to main.scss so it isn't empty
- Adds node and building of front-end assets to docker

### How to review
- In a local terminal, run `nvm use`, then `npm ci`, then `npm run build`
- Make some changes to `main.js` and `main.scss`.
- Re-run `npm run build` and test the changes on any page on the site
- Run `npm run lint` to test the linting
- Run `npm run start` to test watch mode.
- Run `npm run format` to test prettier
- Try modifying `main.scss` to add `color:red` to the `body` rule and then try committing the change - it should fail.
